sdk/identity/azure-identity/src/main/java/com/azure/identity/SharedTokenCacheCredential.java

```diff
@@ -63,92 +51,18 @@
-                Map<String, IAccount> accounts = new HashMap<>(); // home account id -> account
-
-                for (IAccount cached : set) {
-                    if (username == null || username.equals(cached.username())) {
-                        if (!accounts.containsKey(cached.homeAccountId())) { // only put the first one
-                            accounts.put(cached.homeAccountId(), cached);
-                        }
-                    }
-                }
-
-                if (set.size() == 0) {
-                    return Mono.error(new CredentialUnavailableException("SharedTokenCacheCredential authentication "
-                             + "unavailable. No accounts were found in the cache."));
-                }
-
-                if (CoreUtils.isNullOrEmpty(username)) {
-                    return Mono.error(new CredentialUnavailableException("SharedTokenCacheCredential authentication "
-                             + "unavailable. Multiple accounts were found in the cache. Use username and tenant id "
-                             + "to disambiguate."));
-                }
-
-                if (accounts.size() != 1) {
-                    if (accounts.size() == 0) {
-                        return Mono.error(new CredentialUnavailableException(
-                            String.format("SharedTokenCacheCredential authentication "
-                             + "unavailable. No account matching the specified username %s was found in "
-                             + "the cache.", username)));
-                    } else {
-                        return Mono.error(new CredentialUnavailableException(String.format("SharedTokenCacheCredential"
-                             + " authentication unavailable. Multiple accounts matching the specified username %s were "
-                             + "found in the cache.", username)));
-                    }
-                }
-
-                requestedAccount = accounts.values().iterator().next();
-
-
-                // if it does, then request the token
-                SilentParameters params = SilentParameters.builder(
-                        new HashSet<>(request.getScopes()), requestedAccount)
-                    .authorityUrl(authorityUrl)
-                    .build();
-
-                CompletableFuture<IAuthenticationResult> future;
-                try {
-                    future = pubClient.acquireTokenSilently(params);
-                    return Mono.fromFuture(() -> future).map(result ->
-                        new AccessToken(result.accessToken(),
-                            result.expiresOnDate().toInstant().atOffset(ZoneOffset.UTC)));
-
-                } catch (MalformedURLException e) {
-                    e.printStackTrace();
-                    return Mono.error(new RuntimeException("Token was not found"));
-                }
-            });
+        return identityClient.authenticateWithSharedTokenCache(request, username);
```

Comment:
`authenticateWithSharedTokenCache` searches the shared token cache for an account each time. Shouldn't the account be saved off after the first time this method is called so that `authenticateWithMsalAccount` can be called instead?

Reply:
added the logic.